### PR TITLE
Fixing the branch protection issue.

### PR DESCRIPTION
**Pull request by ravali-rimmalapudi**
# Fixes #

Closes [DII-134](https://issues.corp.twilio.com/browse/DII-134): Fixing the protected branch issue.
Workflow Runs: [Failed Run](https://github.com/ravali-rimmalapudi/twilio-cli-core/runs/3728882137?check_suite_focus=true#step:6:66), [Passed Run](https://github.com/ravali-rimmalapudi/twilio-cli-core/runs/3729045192?check_suite_focus=true#step:6:64)
BranchProtection: [Enabled](https://github.com/ravali-rimmalapudi/twilio-cli-core/settings/branches)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ jobs:
     steps:
       - name: Checkout cli
         uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.REPO_ACCESS_TOKEN }}
```
Comment (shrutiburman) [inline .github/workflows/release.yml:42]: Is it a custom token variable? Where will this value come from?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -49,6 +51,8 @@ jobs:
     steps:
       - name: Checkout cli
         uses: actions/checkout@v2
+        with:
+          persist-credentials: false
```
Comment (shrutiburman) [inline .github/workflows/release.yml:55]: Is this to stop persisting the auth tokens? Is it by-default set to `true`? If yes, don't we need to have this at all places?

File: .github/workflows/release.yml (hunk 1 above)
Comment (ravali-rimmalapudi) [reply .github/workflows/release.yml:42]: yes, it is from [secrets](https://github.com/twilio/twilio-cli/settings/secrets/actions).

**Review by onuzbee — APPROVED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (shrutiburman) [reply .github/workflows/release.yml:55]: Okay. 
I meant do we need to add this `persist-credentials: false` everywhere else for security reasons, maybe in a cleanup stage? Just trying to ensure that the token is not valid after the job is done.

**Review by shrutiburman — APPROVED** [review]
🚀

File: .github/workflows/release.yml (hunk 2 above)
Comment (ravali-rimmalapudi) [reply .github/workflows/release.yml:55]: As per the [docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication), **Before each job begins, GitHub fetches a token for the job. The token expires when the job is finished.**.

**Pull request merged**